microsoft-edge/webview2/concepts/working-with-local-content.md

```diff
@@ -0,0 +1,627 @@
+* [CoreWebView2.WebResourceRequested Event](/dotnet/api/microsoft.web.webview2.core.corewebview2.webresourcerequested)
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+* [CoreWebView2.NavigateWithWebResourceRequest Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#navigatewithwebresourcerequest)
+* [CoreWebView2.WebResourceRequested Event](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#webresourcerequested)
+
+##### [Win32/C++](#tab/win32cpp)
+
+* [ICoreWebView2_2::NavigateWithWebResourceRequest method](/microsoft-edge/webview2/reference/win32/icorewebview2_2#navigatewithwebresourcerequest)
+* [ICoreWebView2::WebResourceRequested event (add](/microsoft-edge/webview2/reference/win32/icorewebview2#add_webresourcerequested), [remove)](/microsoft-edge/webview2/reference/win32/icorewebview2#remove_webresourcerequested)
+
+---
+
+
+<!-- ------------------------------ -->
+#### Example of handling the WebResourceRequested event
+
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+```csharp
+// Reading of response content stream happens asynchronously, and WebView2 does not 
+// directly dispose the stream once it read.  Therefore, use the following stream
+// class, which properly disposes when WebView2 has read all data.  For details, see
+// [CoreWebView2 does not close stream content](https://github.com/MicrosoftEdge/WebView2Feedback/issues/2513).
+class ManagedStream : Stream {
+    public ManagedStream(Stream s)
+    {
+        s_ = s;
+    }
+
+    public override bool CanRead => s_.CanRead;
+
+    public override bool CanSeek => s_.CanSeek;
+
+    public override bool CanWrite => s_.CanWrite;
+
+    public override long Length => s_.Length;
+
+    public override long Position { get => s_.Position; set => s_.Position = value; }
+
+    public override void Flush()
+    {
+        throw new NotImplementedException();
+    }
+
+    public override long Seek(long offset, SeekOrigin origin)
+    {
+        return s_.Seek(offset, origin);
+    }
+
+    public override void SetLength(long value)
+    {
+        throw new NotImplementedException();
+    }
+
+    public override int Read(byte[] buffer, int offset, int count)
+    {
+        int read = 0;
+        try
+        {
+            read = s_.Read(buffer, offset, count);
+            if (read == 0)
+            {
+                s_.Dispose();
+            }
+        } catch (Exception e)
+        {
+            s_.Dispose();
+            throw e;
+        }
+        return read;
+    }
+
+    public override void Write(byte[] buffer, int offset, int count)
+    {
+        throw new NotImplementedException();
+    }
+
+   private Stream s_;
+}
+webView.CoreWebView2.AddWebResourceRequestedFilter("https://demo/*", 
+                                                CoreWebView2WebResourceContext.All);
+webView.CoreWebView2.WebResourceRequested += delegate (object sender, 
```

Comment:
@yildirimcagri - Pls techreview my [commit](https://github.com/MicrosoftDocs/edge-developer/pull/2353/commits/ba4e4d0eb765bce13cf53b567d04225dc76396f0) in which I wrapped code at column 85.